apps/frontend/components/repository/Outline/CopyActivity/index.vue

```diff
@@ -0,0 +1,193 @@
+import Promise from 'bluebird';
+import { SCHEMAS } from 'tailor-config-shared';
+import sortBy from 'lodash/sortBy';
+import { useLoader } from '@tailor-cms/core-components-next';
+
+import { activity as activityApi, repository as repositoryApi } from '@/api';
+import type { Activity } from '@/api/interfaces/activity';
+import type { Repository } from '@/api/interfaces/repository';
+import RepositoryTree from './RepositoryTree.vue';
+import TailorDialog from '@/components/common/TailorDialog.vue';
+import { useActivityStore } from '@/stores/activity';
+import { useCurrentRepository } from '@/stores/current-repository';
+
+interface Props {
+  repositoryId: number;
+  levels: number[];
+  action: string;
+  anchor?: Activity | null;
+  showActivator?: boolean;
+}
+
+const props = withDefaults(defineProps<Props>(), {
+  anchor: null,
+  showActivator: false,
+});
+const emit = defineEmits(['close', 'completed']);
+
+const store = useCurrentRepository();
+const activityStore = useActivityStore();
+const { loading: isFetchingRepositories, loader } = useLoader();
+
+const visible = ref(!props.showActivator);
+const repositories = ref<Repository[]>([]);
+const selectedRepository = ref<Repository | null>(null);
+const selectedActivities = ref<Activity[]>([]);
+const copiedActivities = ref<Activity[]>([]);
+const isFetchingActivities = ref(false);
+const isCopyingActivities = ref(false);
+
+const schema = computed(() =>
+  SCHEMAS.find((it) => store.repository?.schema === it.id),
+);
+
+const copyBtnLabel = computed(() => {
+  const selectedCount = selectedActivities.value.length;
+  const selectionLabel = selectedCount
+    ? `${pluralize('item', selectedCount, true)}`
+    : '';
+  return `Copy ${selectionLabel} ${
+    props.action === InsertLocation.ADD_INTO ? 'inside' : ''
+  }`;
+});
+
+const selectRepository = async (repository: Repository) => {
+  if (!repositories.value.find(({ id }) => id === repository?.id)) return;
+  selectedRepository.value = repository;
+  selectedActivities.value = [];
+  if (repository.activities) return;
+  isFetchingActivities.value = true;
+  const activities = await activityApi.getActivities(repository.id);
+  repository.activities = sortBy(activities, 'position');
+  isFetchingActivities.value = false;
+};
+
+const copyActivity = async (activity: Activity) => {
+  const { id: srcId, repositoryId: srcRepositoryId, type } = activity;
+  const { action } = props;
+  const anchor =
+    (action === InsertLocation.ADD_AFTER && last(copiedActivities.value)) ||
+    props.anchor;
+  const payload = {
+    srcId,
+    srcRepositoryId,
+    repositoryId: props.repositoryId,
+    type,
+    position: await activityStore.calculateCopyPosition(
+      action,
+      anchor as Activity,
+    ),
+  } as any;
+  if (anchor) {
+    payload.parentId =
+      action === InsertLocation.ADD_INTO ? anchor.id : anchor.parentId;
+  }
+  const activities = await activityStore.clone(payload);
+  copiedActivities.value.push(activities[0]);
```

Comment:
Can we avoid causing side-effects like this (mutating outside scope variable without explicit assign upon return). Might make sense to add comment why only first one.